bin/cle-test

```diff
@@ -0,0 +1,349 @@
+
+    return $timeout_seconds;
+}
+
+sub get_process_timeout {
+    my $timeout_hours = 14;
+    my $timeout_seconds = $timeout_hours * 3600;
+
+    return $timeout_seconds;
+}
+
+sub wait_for_build {
+    my $build = shift;
+    my $start_time = shift;
+    my $event = $build->the_master_event;
+    unless ($event) {
+        fail("Could not get the build's master event!\n");
+    }
+
+    my $timeout = get_timeout();
+    printf("Monitoring build (%s) until it completes or timeout "
+        . "of %s minutes is reached.\n\n", $build->id, $timeout / 60);
+
+    while (!grep { $event->event_status eq $_ } ('Succeeded',
+            'Failed', 'Crashed')) {
+        UR::Context->current->reload($event);
+        $build = UR::Context->current->reload('Genome::Model::Build', id => $build->id);
+
+        my $elapsed_time = time - $start_time;
+        if ($elapsed_time > $timeout) {
+            printf("Build (%s) timed out after %s minutes",
+                $build->id, $timeout / 60);
+            Library::send_timeout_mail();
+            build_view_and_exit($build);
+        }
+
+        sleep(30);
+    }
+}
+
+sub wait_for_process {
+    my $process = shift;
+
+    my $timeout = get_process_timeout();
+    printf("Monitoring process (%s) until it completes or timeout "
+        . "of %s minutes is reached.\n\n", $process->id, $timeout / 60);
+
+    my $start_time = time;
+    while (!grep { $process->status eq $_ } ('Succeeded', 'Crashed')) {
+        UR::Context->current->reload($process);
+
+        my $elapsed_time = time - $start_time;
+        if ($elapsed_time > $timeout) {
+            printf("Process (%s) timed out after %s minutes",
+                $process->id, $timeout / 60);
+            Library::send_timeout_mail();
+            process_view_and_exit($process);
+        }
+
+        sleep(30);
+    }
+}
+
+sub process_view_and_exit {
+    my $process = shift;
+    my $pv_command = Genome::Process::Command::View->create(
+        process => $process);
+    $pv_command->execute;
+    exit(255);
+}
+
+sub check_build_failure {
+    my $build = shift;
+
+    if ($build->status eq 'Succeeded') {
+        printf("Build status is %s.\n", $build->status);
+    } else {
+        Library::send_fail_mail();
+        build_view_and_exit($build);
+    }
+}
+
+sub diff_build {
+    my $build = shift;
+
+    printf('Starting diff (new build = %s)...', $build->id);
+
+    my @blessed_builds = Genome::Model::Build->get(id => [qw(
+    185d8bac3d7c4437b7ce9207dfadac7e
+    5f57f08a8fd84886b275f0b5571e9fd7
+    e02e8a5ccaad458d839de51eb47d8d2c
+    26e65adaa8034dd99ef92b27f61ad862
+    3243f261a8b64c089a5254291f7c2de3
+    f87701c292e843958d088e171a65a67a
+    301d5d51c96e41308d015008720f3962
+    )]);
+
+    my $matching_blessed_build;
+    for my $blessed_build (@blessed_builds) {
+        if ($build->tumor_sample eq $blessed_build->tumor_sample and
+                (($build->normal_sample == undef and $blessed_build->normal_sample == undef) or
```

Comment:
The `undef` checks could be written as: `!defined($build->normal_sample) and !defined($blessed_build->normal_sample)`.